staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types_jsonschema.go

```diff
@@ -55,6 +55,36 @@ type JSONSchemaProps struct {
 	Definitions          JSONSchemaDefinitions
 	ExternalDocs         *ExternalDocumentation
 	Example              *JSON
+
+	// x-kubernetes-preserve-unknown-fields stops the API server
+	// decoding step from pruning fields which are not specified
+	// in the validation schema. This affects fields recursively,
+	// but switches back to normal pruning behaviour if nested
+	// properties or additionalProperties are specified in the schema.
+	XPreserveUnknownFields bool
```

Comment:
curious if you considered grouping these inside an inline `Extensions` field here. I'm fine with either approach, just wondered

Reply:
no, didn't consider it. They are not reused, not sure it adds anything.